tests/gordo_components/data_provider/test_data_provider_iroc.py

```diff
@@ -105,6 +122,32 @@ def test_load_series_no_tag_list(self):
         )
         self.assertEqual([], res)
 
+    def test_load_series_many_assets(self, _mocked_method):
+        """load_series will return an empty generator when called with tags
+        related to several assets"""
+        iroc_reader = IrocReader(client=None, threads=1)
+        res = list(
+            iroc_reader.load_series(
+                from_ts=isoparse("2018-05-02T01:56:00+00:00"),
+                to_ts=isoparse("2018-05-03T01:56:00+00:00"),
+                tag_list=IROC_MANY_ASSETS_TAG_LIST,
+            )
+        )
+        self.assertEqual([], res)
+
+    def test_load_series_no_asset_found(self, _mocked_method):
+        """load_series will return an empty generator when called with tags
+        that cannot be related to any asset"""
+        iroc_reader = IrocReader(client=None, threads=1)
+        res = list(
+            iroc_reader.load_series(
+                from_ts=isoparse("2018-05-02T01:56:00+00:00"),
+                to_ts=isoparse("2018-05-03T01:56:00+00:00"),
+                tag_list=IROC_NO_ASSET_TAG_LIST,
+            )
+        )
+        self.assertEqual([], res)
+
```

Comment:
My preference would actually be that these tests caused an exception to be raised - since it is non-supported input, and we might as well fail hard.